UpstreamDownstreamDialog.py

```diff
@@ -82,3 +126,7 @@ def __addLayer(self, url, title):
         layer = QgsRasterLayer("url=%s&crs=EPSG:2193&format=image/png"
                                % (url % hydroid), "%s %s" % (title, hydroid), "wms")
         QgsMapLayerRegistry.instance().addMapLayer(layer)
+    
+    def __onProjectRead(self):
+            reg = QgsMapLayerRegistry.instance()
+            self.recLayer = reg.mapLayer(QgsProject.instance().readEntry("fishdb", "recLayer", "")[0])
```

Comment:
Did you check whether this does not raise an exception if the setting does not exist in the project? (I.e. accessing `[0]` of a possibly `None` return value.)